bookwyrm/utils/isni.py

```diff
@@ -0,0 +1,152 @@
+"""ISNI author checking utilities"""
+import xml.etree.ElementTree as ET
+import requests
+
+
+def url_stringify(string):
+    """replace spaces for url encoding"""
+
+    # TODO: this is very lazy and incomplete
+    return string.replace(" ", "+")
+
+
+def request_isni_data(search_index, search_term, max_records=5):
+    """Request data from the ISNI API"""
+
+    search_string = url_stringify(search_term)
+    query_parts = [
+        "http://isni.oclc.org/sru/?query=",
+        search_index,
+        "+%3D+%22",
+        search_string,
+        "%22&version=1.1&operation=searchRetrieve&recordSchema=isni-b",
+        "&maximumRecords=",
+        str(max_records),
+        "&startRecord=1&recordPacking=xml&sortKeys=RLV%2Cpica%2C0%2C%2C",
+    ]
```

Comment:
[There's an easier way!](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlencode) You can generate the url by passing a dict of the query params into urlencode

Reply:
whoops how embarrasing. I meant to clean this up and forgot about it: the plan for this whole PR was "I wonder if this would even work? Let's find out!" hence the trail of mess I left behind me 🤣

Reply:
LOL actually there is an even easier way because `requests` takes a params dict and url encodes the values.